scripts/lib/CIME/case/case.py

```diff
@@ -84,8 +84,7 @@ def __init__(self, case_root=None, read_only=True, record=False):
 
         if case_root is None:
             case_root = os.getcwd()
-        if not (in_doctest() or in_unit_test()):
```

Comment:
A bit of explanation for this change and the related changes in this file – doing away with the recently-added in_doctest() and in_unit_test() functions: I have replaced these functions with explicit handling in the unit tests and doctests that needed it, either by ensuring that caseroot is a non-existent directory or by putting an empty `env_case.xml` file in the caseroot directory. Reasons I prefer this method: (1) (main reason) it works for `python -m unittest discover`; (2) I found the other method hard to understand; (3) it makes me nervous to have unit test-specific code sprinkled in the production code, especially when that test-specific code is hard to understand: it's hard to be confident that things will work right in production.